arch/risc-v/src/common/supervisor/riscv_sbi.c

```diff
@@ -87,13 +93,17 @@ static inline uintptr_t sbi_ecall(unsigned int extid, unsigned int fid,
 
 void riscv_sbi_set_timer(uint64_t stime_value)
 {
+#ifdef CONFIG_NUTTSBI
+  sbi_mcall_set_timer(stime_value);
```

Comment:
can we make nuttx sbi use the same ecall interface?

Reply:
It is not trivial, I think this level of abstraction is good enough.

`sbi_mcall_set_timer` implements / uses the custom ecall interface. The reason is, I don't want to make any expectations that the NuttX native SBI will ever conform to OpenSBI specification.

I still strongly suggest using OpenSBI, but integrating it can be quite a burden. This is why I made this simple / minimal SBI, because it made (and makes) developing the kernel mode much faster.